### PR TITLE
Smart contract IT added and code correction for fetching sequence and package id for chaincode

### PR DESCRIPTION
**Pull request by abhakish**
This PR includes :
1. Smart contract IT and sample chaincode for deployment
2. Code correction for fetching sequence and package id for chaincode
3. Removed unnecessary files from fabric setup
4. Code correction to throw exception if channel not found before fetching channel members

Signed-off-by: Abhay Kishore <abhay.kishore@walmart.com>

### CODE REVIEW

File: src/main/java/hlf/java/rest/client/service/impl/ChaincodeOperationsServiceImpl.java [hunk 1]
```diff
@@ -94,26 +96,29 @@ public String getCurrentSequence(
...
+        if (response.getStatus() == ProposalResponse.Status.SUCCESS
+            && response.getVersion().equals(chaincodeVersion)) {
+          sequence = response.getSequence();
+          break;
+        } else {
+          if (404 == response.getChaincodeActionResponseStatus()) {
+            // not found set sequence to 1;
+            sequence = 1;
+            break;
+          }
```
Comment (adityajoshi12) [inline src/main/java/hlf/java/rest/client/service/impl/ChaincodeOperationsServiceImpl.java:111]: [nitpick] : Use constant HttpStatus.NOT_FOUND
Q: can there be other scenerio like with status code 500?

File: src/main/java/hlf/java/rest/client/service/impl/ChaincodeOperationsServiceImpl.java [hunk 2]
```diff
@@ -94,26 +96,29 @@ public String getCurrentSequence(
...
         }
       }
 
-      if (sequenceNumbers.size() == 0) {
+      if (sequence < 0) {
```
Comment (adityajoshi12) [inline src/main/java/hlf/java/rest/client/service/impl/ChaincodeOperationsServiceImpl.java:115]: [nitpick]: consider strict checking (sequence == -1)

File: src/main/java/hlf/java/rest/client/service/impl/ChaincodeOperationsServiceImpl.java (hunk 1 above)
Comment (abhakish) [reply src/main/java/hlf/java/rest/client/service/impl/ChaincodeOperationsServiceImpl.java]: 1. Constant added.
2. We only what to check for scenario where given chaincode name isn't found i.e 404.

File: src/main/java/hlf/java/rest/client/service/impl/ChaincodeOperationsServiceImpl.java (hunk 2 above)
Comment (abhakish) [reply src/main/java/hlf/java/rest/client/service/impl/ChaincodeOperationsServiceImpl.java]: done

File: src/main/java/hlf/java/rest/client/service/impl/ChaincodeOperationsServiceImpl.java [hunk 3]
```diff
@@ -94,26 +96,29 @@ public String getCurrentSequence(
           queryChaincodeDefinitionProposalResponses =
               channel.lifecycleQueryChaincodeDefinition(
                   queryLifecycleQueryChaincodeDefinitionRequest, peers);
-      Set<Long> sequenceNumbers = new HashSet<>();
+      long sequence = -1L;
       for (LifecycleQueryChaincodeDefinitionProposalResponse response :
           queryChaincodeDefinitionProposalResponses) {
-        if (response.getVersion().equals(chaincodeVersion)) {
-          sequenceNumbers.add(response.getSequence());
+        if (response.getStatus() == ProposalResponse.Status.SUCCESS
```
Comment (nithin-pankaj) [inline src/main/java/hlf/java/rest/client/service/impl/ChaincodeOperationsServiceImpl.java:102]: [nitpick] always prefer equals() for checking value equality. This if block does equality check in two different ways.

**Review by nithin-pankaj — APPROVED** [review]

File: src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java [hunk 4]
```diff
@@ -143,6 +143,11 @@ public Set<String> getChannelMembersMSPID(String channelName) {
     }
     Network network = gateway.getNetwork(channelName);
     Channel channel = network.getChannel();
+    try {
+      channel.getChannelConfigurationBytes();
```
[the remark this answers is not in the document]
Comment (abhakish) [reply src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java:147]: On call of this method we get an ProposalException if channel doesn't exist, therefore, return isn't required here

[the remark this answers is not in the document]
Comment (nithin-pankaj) [reply src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java:147]: I meant to add the return statement in line number 151 after this statement.

File: src/main/java/hlf/java/rest/client/service/impl/ChaincodeOperationsServiceImpl.java [hunk 5]
```diff
@@ -94,26 +96,29 @@ public String getCurrentSequence(
...
+            && response.getVersion().equals(chaincodeVersion)) {
+          sequence = response.getSequence();
+          break;
+        } else {
```
[the remark this answers is not in the document]
Comment (abhakish) [reply src/main/java/hlf/java/rest/client/service/impl/ChaincodeOperationsServiceImpl.java]: change done

File: src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (abhakish) [reply src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java:147]: change done

File: src/main/java/hlf/java/rest/client/service/impl/ChaincodeOperationsServiceImpl.java (hunk 3 above)
Comment (abhakish) [reply src/main/java/hlf/java/rest/client/service/impl/ChaincodeOperationsServiceImpl.java]: change done

**Review by arsulegai — APPROVED** [review]

**Pull request merged**